Modules/posixmodule.c

```diff
@@ -11420,14 +11421,50 @@ os_lseek_impl(PyObject *module, int fd, Py_off_t position, int how)
     Py_BEGIN_ALLOW_THREADS
     _Py_BEGIN_SUPPRESS_IPH
 #ifdef MS_WINDOWS
-    result = _lseeki64(fd, position, how);
+    switch (how) {
+    case SEEK_SET:
+        how = FILE_BEGIN;
+        break;
+    case SEEK_CUR:
+        how = FILE_CURRENT;
+        break;
+    case SEEK_END:
+        how = FILE_END;
+        break;
+    }
+    HANDLE h = (HANDLE)_get_osfhandle(fd);
+    if (h == INVALID_HANDLE_VALUE) {
+        result = -1;
+    }
+    if (result >= 0) {
+        if (GetFileType(h) != FILE_TYPE_DISK) {
+            // Only file is seekable
+            errno = ESPIPE;
+            result = -1;
+        }
+    }
+    if (result >= 0) {
+        LARGE_INTEGER distance, newdistance;
+        distance.QuadPart = position;
+        if (SetFilePointerEx(h, distance, &newdistance, how)) {
+            result = newdistance.QuadPart;
+        } else {
+            result = -1;
+        }
+    }
```

Comment:
Why not simply use `_lseeki64()` after checking `GetFileType()`? It looks much simpler.

Reply:
I thought Windows's CRT will call `_get_osfhandle` to convert the POSIX's *fd* to Windows's *handle* internally, which we've already called, so use `SetFilePointerEx` with the *handle* directly will gain some performance benefit. But I run the small benchmark script which mentioned at the top of this PR, and didn't see any noticeable performance regret, so updated to use `_lseeki64` directly.